tests/testGetEimage.py

```diff
@@ -0,0 +1,97 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test ability to get an eimage from the butler.
+"""
+import os
+import unittest
+
+import numpy as np
+
+import lsst.utils
+import lsst.afw.geom as afwGeom
+import lsst.utils.tests
+import lsst.daf.persistence as dafPersistence
+from lsst.afw.coord import Coord, IcrsCoord, Observatory, Weather
+from lsst.daf.base import DateTime
+
+obsLsstSimDir = lsst.utils.getPackageDir('obs_lsstSim')
+InputDir = os.path.join(obsLsstSimDir, 'tests', 'data')
+
+
+class GetEimageTestCase(lsst.utils.tests.TestCase):
+
+    def testGetEimage(self):
+        """Test getting a eimage from the butler.  Mostly we want to test whether the proper metadata
+        gets attached.  Specifically, we'll check for a Wcs and that the VisitInfo is populated.
+        """
+        # For lsstSim specific reasons, we need to specify the raft and sensor
+        dataId = dict(visit=840, raft='2,2', sensor='1,1')
+        butler = dafPersistence.Butler(InputDir)
+        exposure = butler.get('eimage', dataId=dataId)
+
+        # Test for a Wcs object
+        self.assertIsNotNone(exposure.getWcs())
+
+        # Test VisitInfo for known values for this piece of test data
+        methods_to_test = ['getBoresightAirmass', 'getBoresightAzAlt', 'getBoresightRaDec',
+                           'getBoresightRotAngle', 'getDarkTime', 'getDate', 'getExposureId',
+                           'getExposureTime', 'getObservatory', 'getRotType', 'getWeather']
+
+        test_values = [1.00015190967402, Coord(afwGeom.Point2D(0.0, 89.0), afwGeom.degrees, 2000.),
+                       IcrsCoord(afwGeom.Point2D(53.0091385, -27.4389488), afwGeom.degrees),
+                       afwGeom.Angle(-2.5653486740055484, afwGeom.radians), 30.0,
+                       DateTime("1994-01-02T01:46:59.520000913", DateTime.TAI), 430204, 30.0,
+                       Observatory(afwGeom.Angle(-70.749417, afwGeom.degrees),
+                                   afwGeom.Angle(-30.244633, afwGeom.degrees), 2663), 1,
+                       Weather(20, 69327.64145580001, np.nan)]
+
+        def test_weather(w1, w2):
+            """Test equality of two Weather objects
+            @param[in] w1  First Weather object
+            @param[in] w2  Second Weather object
+            """
+            humid_bool = np.isnan(w1.getHumidity()) and np.isnan(w2.getHumidity())
+            if not humid_bool:
+                humid_bool = (w1.getHumidity() == w2.getHumidity())
+            self.assertTrue(w1.getAirPressure() == w2.getAirPressure() and w1.getAirTemperature() ==
+                            w2.getAirTemperature() and humid_bool)
+
+        ae = self.assertEqual
```

Comment:
Same as you're doing now: you can just list the asserts in this test, one after the other. It'll still bail on the first one it hits (just as it does now), but it'll be much more obvious what it's bailing on.